### PR TITLE
Sql celery ingest fixes

### PR DESCRIPTION
**Pull request by spacemansteve**
Includes Celery based pipeline, sql based adsdata and fixes to the pipeline.

### CODE REVIEW

**Comment by spacemansteve** [comment]
Currently only one test is failing: test_app.py/TestWorkers.test_task_merge_metadata

**Comment by spacemansteve** [comment]
I am confused why I had to change the pubnote test data for one bibcode (https://github.com/spacemansteve/ADSimportpipeline/blob/sqlCeleryIngestFixes/tests/stubdata/ADSRecords/2015ApJ...815..133S#L716) but not another (https://github.com/spacemansteve/ADSimportpipeline/blob/sqlCeleryIngestFixes/tests/stubdata/ADSRecords/2017FrPhy..12l0504C#L512).  I thought the first pubnote was incorrectly formed and changed it (https://github.com/spacemansteve/ADSimportpipeline/commit/f539e39acc7fcb2476d92ed9852fb5888fe1c7bb#diff-28c31b4299319c6b60a83c8dee335c79L715) so tests would pass.  I didn't have to change the second one.

File: config.py [hunk 1]
```diff
@@ -8,9 +8,29 @@
...
-  'http://ads1/solr/update',
-  'http://ads2/solr/update',
+  'http://localhost:9983/solr/BumblebeeETL/update' 
 ]
+#  'http://ads1/solr/update',
+#  'http://ads2/solr/update',
+
+
+
+SQL = {
```
Comment (romanchyla) [inline config.py]: this is a small point, but worth considering: use 'dialect://user:pass@host/db' format - sqlalchemy can parse it

File: lib/SqlConnection.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+from sqlalchemy.dialects.postgresql import ARRAY
+
+class PipelineSqlConnection:
+
+    def __init__(self,**kwargs):
+        self.logger = kwargs.get('logger',None)
+        if not self.logger:
+            self.initializeLogging(**kwargs)
+
+        self.host = kwargs.get('HOST','localhost')
```
Comment (romanchyla) [inline lib/SqlConnection.py]:
```
from sqlalchemy import engine
u = engine.url.make_url('postgres://duser@localhost/workloadmodel')
print u.host
```

prints 'localhost'

File: lib/SqlConnection.py [hunk 3]
```diff
@@ -0,0 +1,149 @@
+import os,sys
+import logging
+import logging.handlers
+from cloghandler import ConcurrentRotatingFileHandler
+from collections import deque
+from sqlalchemy import *
```
Comment (romanchyla) [inline lib/SqlConnection.py]: please use explicit imports, not '*'

File: lib/SqlConnection.py [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+                     schema=self.schema,
+                     extend_existing=True)
+
+
+    def getAllBibcodes(self):
+        return_value = deque()
+        row_view = self.get_row_view_table()
+        s = select([row_view])
+        rp = self.conn.execute(s)
+        rows = rp.fetchall()
```
Comment (romanchyla) [inline lib/SqlConnection.py]: how many recs (in the worst case) can this operation fetch?

File: tests/tests/readrecords.py [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+import os
+import sys
+import json
+import codecs
+import argparse
+
+PROJECT_HOME = os.path.abspath(os.path.join(os.path.dirname(__file__),'../../'))
+sys.path.insert(0, PROJECT_HOME)
+
+from lib import ReadRecords
```
Comment (romanchyla) [inline tests/tests/readrecords.py]: those libs don't exist anymore (from aip.libs import read_records)

File: tests/tests/testSql.py [hunk 6]
```diff
@@ -0,0 +1,85 @@
+import sys, os
+
+PROJECT_HOME = os.path.abspath(os.path.join(os.path.dirname(__file__), '../../'))
+sys.path.append(PROJECT_HOME)
+from settings import SQL_ADSDATA
+from lib import SqlConnection
```
Comment (romanchyla) [inline tests/tests/testSql.py]: dtto

File: tests/tests/testSql.py [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+from stubdata import stubdata
+import json
+import pymongo
```
Comment (romanchyla) [inline tests/tests/testSql.py]: ...and mongo is gone as well; perhaps remove this script?

File: config.py (hunk 1 above)
Comment (spacemansteve) [inline config.py]: Makes sense.  I had copied the old mongo configuration which provided for default values from an inherited object.  It's a lot of bother for such a small thing.

File: lib/SqlConnection.py (hunk 3 above)
Comment (spacemansteve) [inline lib/SqlConnection.py]: agreed

File: lib/SqlConnection.py (hunk 2 above)
Comment (spacemansteve) [inline lib/SqlConnection.py]: I modified initialization of the engine to match the uri in the config file but still rely on create_engine which accepts a string parameter.  create_engine does not require a url object.

File: lib/SqlConnection.py (hunk 4 above)
Comment (spacemansteve) [inline lib/SqlConnection.py]: worst case is 12m, I added a docstring saying it is only called by test code to verify the sql table was initialized with test data.

Comment (romanchyla) [inline lib/SqlConnection.py]: but even that is not necessary, you can accomplish the same with:

```
rp = self.conn.execute(s)
for row in rp:
   yield row
```

very simple change (if the upstream code can be happy with streamed data)

**Pull request merged**